WooCommerce/Classes/ViewRelated/Themes/ThemesPreviewView.swift

```diff
@@ -0,0 +1,207 @@
+import SwiftUI
+import struct Yosemite.WordPressTheme
+
+/// View to preview the demo page of a WordPress theme.
+/// It lets merchants to:
+/// - Preview the layouts responsively between mobile, tablet, and desktop view.
+/// - Activate the previewed theme on the current site.
+///
+struct ThemesPreviewView: View {
+    enum PreviewDevice: CaseIterable, Identifiable {
+        var id: PreviewDevice { self }
+
+        case mobile
+        case tablet
+        case desktop
+
+
+        var menuTitle: String {
+            switch self {
+            case .desktop:
+                return ThemesPreviewView.Localization.menuDesktop
+            case .tablet:
+                return ThemesPreviewView.Localization.menuTablet
+            case .mobile:
+                return ThemesPreviewView.Localization.menuMobile
+            }
+        }
+
+        /// The initial layout used as preview.
+        static var defaultDevice: PreviewDevice {
+            return UIDevice.current.userInterfaceIdiom == .pad ? .tablet : .mobile
+        }
+
+        /// The width is used in the `viewportScript` JS to change the WebView's viewport.
+        /// A theme's CSS will take into account this number to decide whether to responsively display its layout to be
+        /// the mobile, tablet, or desktop layout.
+        ///
+        var browserWidth: CGFloat {
+            switch self {
+            case .mobile: return 400
+            case .tablet: return 800
+            case .desktop: return 1200
+            }
+        }
+
+        /// This JavaScript forces the WebView's viewport to match the supplied width above. This allows the preview
+        /// functionality to switch between a theme's mobile, tablet, or desktop layout.
+        ///
+        var viewportScript: String {
+            let js = """
+            // remove all existing viewport meta tags - some themes included multiple, which is invalid
+            document.querySelectorAll("meta[name=viewport]").forEach( e => e.remove() );
+            // create our new meta element
+            const viewportMeta = document.createElement("meta");
+            viewportMeta.name = "viewport";
+            viewportMeta.content = "width=%1$d";
+            // insert the correct viewport meta tag
+            document.getElementsByTagName("head")[0].append(viewportMeta);
+            """
+
+            return String(format: js, NSInteger(browserWidth))
+        }
+    }
+
+    @State private var selectedDevice: PreviewDevice = PreviewDevice.defaultDevice
+    private let theme: WordPressTheme
+
+    /// Triggered when "Start with this theme" button is tapped.
+    var onStart: () -> Void
+
+    init(theme: WordPressTheme, onStart: @escaping () -> Void) {
+        self.theme = theme
+        self.onStart = onStart
+    }
+
+    var body: some View {
+        NavigationView {
+            VStack(spacing: 0) {
+                if let url = theme.themeThumbnailURL {
```

Comment:
Do you mean using the demo link instead?

Reply:
Great catch, thanks https://github.com/woocommerce/woocommerce-ios/pull/11452/commits/98047c1f6d305597ae3cf9b31d0b5c2853d68c36